pkg/project/config.go

```diff
@@ -0,0 +1,109 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/wakatime/wakatime-cli/pkg/log"
+
+	"github.com/spf13/viper"
+	"gopkg.in/ini.v1"
+)
+
+const (
+	// WakaTimeConfigFile is the new INI config file which contains project settings
+	// and can override global configuration per project.
+	WakaTimeConfigFile = ".wakatime"
+)
+
+// ConfigDetector detects project configuration from .wakatime files.
+type ConfigDetector struct {
+	Filepath string
+}
+
+// ConfigResult contains both project information and configuration data.
+type ConfigResult struct {
+	Result
+	Config *viper.Viper
+	Found  bool
+}
+
+// DetectConfig finds project configuration from .wakatime INI files.
+// It reads the full configuration file and returns both project info and config.
+func (c ConfigDetector) DetectConfig(ctx context.Context) (ConfigResult, error) {
+	fp, found := FindFileOrDirectory(ctx, c.Filepath, WakaTimeConfigFile)
+	if !found {
+		return ConfigResult{Found: false}, nil
+	}
+
+	logger := log.Extract(ctx)
+	logger.Debugf("wakatime config file found at: %s", fp)
+
+	// Load the INI file
+	cfg, err := ini.LoadSources(ini.LoadOptions{
+		AllowPythonMultilineValues: true,
+		SkipUnrecognizableLines:    true,
+	}, fp)
+	if err != nil {
```

Comment:
My guess is this will always error when it finds the `~/.wakatime` folder and tries to read the folder as a project config file.